utils/misc/misc.go

```diff
@@ -296,12 +311,39 @@ func RemoveFilePaths(filePaths ...string) {
 	}
 }
 
+// GetTemporaryFolderPaths returns all temporary folder paths.
+func GetTemporaryFolderPaths() (tempFolders map[string]bool) {
+	tmpDirPath, err := CreateTMPDIR()
+	if err != nil {
+		return
+	}
+
+	tempFolders = make(map[string]bool)
+	tempFolders[tmpDirPath] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderAsyncDestinationLogs)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderArchives)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderWarehouseStagingUploads)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderRawDataDestinationLogs)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderWarehouseLoadUploadsTmp)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderIdentityMergeRulesTmp)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderIdentityMappingsTmp)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderRedshiftManifests)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, RudderWarehouseJsonUploadsTmp)] = true
+	tempFolders[fmt.Sprintf(`%s/%s`, tmpDirPath, config.GetEnv("RUDDER_CONNECTION_TESTING_BUCKET_FOLDER_NAME", RudderTestPayload))] = true
+	return
+}
+
 // RemoveEmptyFolderStructureForFilePath recursively cleans up everything till it reaches the stage where the folders are not empty or parent.
 func RemoveEmptyFolderStructureForFilePath(fp string) {
 	if fp == "" {
 		return
 	}
 	for currDir := filepath.Dir(fp); currDir != "/" && currDir != "."; {
+		// Checking if the currDir is present in the temporary folders or not
+		// If present we should stop at that point.
+		if _, ok := tempFolders[currDir]; ok {
```

Comment:
Can you rename it to something like reservedFolders/internalFolders?

Reply:
Done.